### PR TITLE
DELETE /playlists/:id/favorites/:id deletes all favorites #73

### PR DESCRIPTION
**Pull request by nathangthomas**
This PR fixes the bug that was previously deleting all playlist_favorites that shared a single favorite.

### CODE REVIEW

File: routes/api/v1/playlistFavorites.js [hunk 1]
```diff
@@ -32,8 +32,9 @@ router.delete('/:favId', async (request, response) => {
       findFavorite(request.params.favId)
       .then(data => {
         if (data) {
-          let targetId = request.params.favId
-          deletePlaylistFavorite(targetId)
+          let favId = request.params.favId
+          let playlistId = request.params.playlistId
```
Comment (fentontaylor) [inline routes/api/v1/playlistFavorites.js]: You could save these as variables earlier to be used in findPlaylist() and findFavorite()

File: tests/playlistFavorites/destroy.spec.js [hunk 2]
```diff
@@ -29,11 +29,17 @@ describe("DELETE /api/v1/playlists/:id/favorites/:id", () => {
...
 
     await database('playlist_favorites')
-      .insert({ id: 2, playlist_id: 1, favorite_id: 2})
+      .insert({ id: 2, playlist_id: 2, favorite_id: 1})
+
+    await database('playlist_favorites')
+      .insert({ id: 3, playlist_id: 2, favorite_id: 2})
 
     var response = await request(app)
       .delete('/api/v1/playlists/1/favorites/1')
```
Comment (fentontaylor) [inline tests/playlistFavorites/destroy.spec.js]: After deleting, you could do
```
let result = await database
  .where({ playlist_id: 2, favorite_id:1 })
  .first()
```
and expect that it returns a record

Comment (nathangthomas) [reply tests/playlistFavorites/destroy.spec.js]: That is a good point.

Comment (nathangthomas) [reply tests/playlistFavorites/destroy.spec.js]: I have made this change.

File: routes/api/v1/playlistFavorites.js (hunk 1 above)
Comment (nathangthomas) [reply routes/api/v1/playlistFavorites.js]: I have made this change.

**Pull request merged**